tests/network/services.go

```diff
@@ -247,67 +232,41 @@ var _ = SIGDescribe("Services", func() {
 		})
 
 		Context("with a service matching the vmi exposed", func() {
-			var jobsToCleanup []func() error
-			var serviceManager *vmiServiceManager
-			var serviceNamePrefix string
-
-			BeforeEach(func() {
-				serviceNamePrefix = "myservice"
-				serviceManager = newVMIServiceManager(inboundVMI, servicePort, serviceNamePrefix, selectorLabelKey, selectorLabelValue)
-
-				services := serviceManager.buildK8sServicesSpec()
-				Expect(services).NotTo(BeEmpty(), "a service should be exposed per each iface. At least one *must* be present.")
+			var jobToCleanup func() error
+			var service *k8sv1.Service
 
-				for _, exposedService := range services {
-					_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(&exposedService)
-					Expect(err).NotTo(HaveOccurred())
-				}
-
-				// reset the job cleanup functions list
-				jobsToCleanup = nil
+			AfterEach(func() {
+				Expect(jobToCleanup()).To(Succeed(), "cleaning up the k8sv1.Job entity should have succeeded.")
 			})
 
-			AfterEach(func() {
-				var errors []error
-				for _, exposedService := range serviceManager.services {
-					err := virtClient.CoreV1().Services(inboundVMI.Namespace).Delete(exposedService.Name, &k8smetav1.DeleteOptions{})
-					errors = append(errors, err)
-				}
-				for _, err := range errors {
-					Expect(err).NotTo(HaveOccurred())
+			table.DescribeTable("should be able to reach the vmi based on labels specified on the vmi", func(ipFamily k8sv1.IPFamily) {
+				serviceName := "myservice"
+				if ipFamily == k8sv1.IPv6Protocol {
+					serviceName = serviceName + "v6"
+					service = buildIPv6ServiceSpec(serviceName, servicePort, servicePort, selectorLabelKey, selectorLabelValue)
+				} else {
+					service = buildServiceSpec(serviceName, servicePort, servicePort, selectorLabelKey, selectorLabelValue)
 				}
-			})
 
-			AfterEach(func() {
-				cleanupJobs(jobsToCleanup)
-			})
+				_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(service)
+				Expect(err).NotTo(HaveOccurred(), "the k8sv1.Service entity should have been created.")
 
-			It("should be able to reach the vmi based on labels specified on the vmi", func() {
-				for _, exposedService := range serviceManager.services {
-					jobCleanupFunc := assertConnectivityToService(exposedService.Name, inboundVMI.Namespace, servicePort)
-					jobsToCleanup = append(jobsToCleanup, jobCleanupFunc)
-				}
-			})
+				jobToCleanup = assertConnectivityToService(serviceName, inboundVMI.Namespace, servicePort)
+
+				Expect(cleanupService(inboundVMI.Namespace, service.Name)).To(Succeed(), "cleaning up the k8sv1.Service entity should have succeeded.")
```

Comment:
As we discussed, please move to the `AfterEach` so the service will be cleaned even if the test failed.

Reply:
Done.